### DISCUSSION TITLE
Project Sponsorship Requests and Updates

### DISCUSSION THREAD

**Discussion by mattsta** [ib-api-reloaded/ib_async#92] [General]
Hey All,

Anybody out there have extra funds to help improve our IBKR API development?

## More Sponsorship == More Development

Just curious if there are any successful people using this library willing to sponsor ongoing feature development of our IBKR API improvements.

We have many hobbyists using this library, but there are probably some people making a couple million a month backed by this library too, right?

I have been trying to keep up with requests and feedback as much as possible, but I've also been between jobs for a while (tech layoff season apparently, nobody is hiring for years) and I don't have the extra time for all the updates/maintaining/reviewing/testing to keep updating everything as quickly as I would like.

Adding more funds would help:

- improve release frequency
- increase the speed of new development
- adapt newer IBKR API features we haven't added yet
- improve documentation
- add more examples of real world usage
- let us explore adapting newer [web API](https://www.interactivebrokers.com/campus/ibkr-api-page/webapi-doc/) features into the current API or maybe even creating a new web api project
- extended auto-trading feature sets with [other projects](https://github.com/mattsta/icli) as just "set it and forget it" automated profit generators trading in and out of positions dozens of times per day with high probability market trend detection logic.

## Funding

Any contributions would be welcome. The best case scenario is having already well-funded people support ongoing development of the tools they are using.

Feel free to use the [built-in GitHub payment feature](https://github.com/sponsors/mattsta) or [just drop me a note](mailto:matt@matt.sh) for more direct recurring sponsorship options (wires/ach/invoicing/apple/etc).

Thanks, and may the software be with you,

-Matt

**Comment by geepeem** [comment]
Wow the response to you plea is positively overwhelming!  /s
I have made a small contribution, will add some more when my trading bots get their act together and start making mega bucks.
Thanks for everything you do Matt - you are appreciated!

**Comment by mattsta** [comment]
ha, thanks! Noticed and very grateful. Every bit helps and is appreciated these days. Overall there's a minimum [survivability](https://en.wikipedia.org/wiki/Survivability) threshold before everything falls apart, so trying to summon more funds from helpful profitable users before worse things kick in.

My own IBKR auto-trading would have probably made a couple million with the volatility this week if it still had enough to run (lots of things were swinging 600% multiple times per day with all these intraday retracements (except for Friday)), but there's not enough funds left to let it rip on its own. Being left out of exponential progress is quite a downer.

**Reply by geepeem** [reply]
Sorry to hear that! Have you thought about offering a course on coding bots using ib_async? I would be interested for sure!
Alternatively, you could apply for funding through TopStep for example..

**Reply by mattsta** [reply]
> Have you thought about offering a course on coding bots using ib_async?

That's an interesting idea, but over the years I've become allergic to heavy effort-up-front projects where the demand ends up being slim.

I think the biggest `ib_async` programming confusion mostly comes from people not understanding how python async programming works, and it can take weeks or months for "how async works" to actually click for some people (it's all just understanding "the event loop", event loop scheduling, `await`, and the difference between coroutines and coroutine functions, but it still takes a while to figure it all out if you're new).

But all ideas are on the table if there's enough interest floating around.

As for TopStep, I've never found a prop firm with API access? I wouldn't want to touch live futures markets without automation. The goal is to get out fast when you're wrong and to stay in long when you're right, and my systems bounce out faster and stay in longer than I can handle mentally between fighting standard MAE inside the ATR, last-chance stop-loss hits, and letting runners stay live for extended amounts of time.

**Reply by geepeem** [reply]
Yeah I get your point about the effort required to set up a course. We're all here trying to develop the holy-grail trading system that will run itself and rake in profits while we lay on a beach somewhere LOL! But in the meantime, we gotta pay the bills somehow.

Are you familiar with QuantConnect? Not sure if you can get direct funding, but they do have a strategy database thingy that you can maybe showcase the performance of your bots and drum up some institutional interest that way... Supposedly you keep the IP rights, but always a risk of someone stealing your code. Just throwing ideas out there!

**Comment by hysech** [comment]
> Hey All,
> 
> Anybody out there have extra funds to help improve our IBKR API development?
> 
> ## More Sponsorship == More Development
> Just curious if there are any successful people using this library willing to sponsor ongoing feature development of our IBKR API improvements.
> 
> We have many hobbyists using this library, but there are probably some people making a couple million a month backed by this library too, right?
> 
> I have been trying to keep up with requests and feedback as much as possible, but I've also been between jobs for a while (tech layoff season apparently, nobody is hiring for years) and I don't have the extra time for all the updates/maintaining/reviewing/testing to keep updating everything as quickly as I would like.
> 
> Adding more funds would help:
> 
> * improve release frequency
> * increase the speed of new development
> * adapt newer IBKR API features we haven't added yet
> * improve documentation
> * add more examples of real world usage
> * let us explore adapting newer [web API](https://www.interactivebrokers.com/campus/ibkr-api-page/webapi-doc/) features into the current API or maybe even creating a new web api project
> * extended auto-trading feature sets with [other projects](https://github.com/mattsta/icli) as just "set it and forget it" automated profit generators trading in and out of positions dozens of times per day with high probability market trend detection logic.
> 
> ## Funding
> Any contributions would be welcome. The best case scenario is having already well-funded people support ongoing development of the tools they are using.
> 
> Feel free to use the [built-in GitHub payment feature](https://github.com/sponsors/mattsta) or [just drop me a note](mailto:matt@matt.sh) for more direct recurring sponsorship options (wires/ach/invoicing/apple/etc).
> 
> Thanks, and may the software be with you,
> 
> -Matt

I appreciate all your commitment and skills you dedicate to. I start my sponsorhip, I sponsored Ewald as well. I am not diving in bucks yet, still develop via paper account. I will start with 50$ and hope to move soon to margin account to upgrade it. Best rergards, Hynek

**Reply by mattsta** [reply]
Thanks so much for the contribution! Every amount helps to free up time resources to build more. Hopefully with a few more funding helpers (or an anonymous whale or two to contribute) we can grow enough of a monthly base where more time can be spent on new features instead of only having a couple hours a week after "normal life stuff" all the time.

I'd like to build-up and release more of my auto-trading platforms because they have oddly good accuracy, but it's a substantial project to refactor all the pieces into "public" working order across multiple systems.

The auto-trading systems use guidelines mostly like this and they often top-tick the right spots (the bars look weird because I scale them by volume, so larger volume bars are wider (and also more important) than smaller volume bars): 
![Screenshot 2024-12-03 at 07 57 04](https://github.com/user-attachments/assets/f0f8bec0-71b7-4f05-ab7f-e4495a82675b)
![Screenshot 2024-12-03 at 08 23 01](https://github.com/user-attachments/assets/6f2c2c09-9ea2-43c2-a80f-3033296acd91)
![Screenshot 2024-12-03 at 08 23 10](https://github.com/user-attachments/assets/7cf169bc-fb38-44d0-a0e4-9dfe480b76fc)
![Screenshot 2024-12-03 at 08 43 14](https://github.com/user-attachments/assets/334c88ad-9d16-4c92-8d83-1f02b019b48e)
![Screenshot 2024-12-03 at 08 56 45](https://github.com/user-attachments/assets/7d8daee1-7374-4a45-8bcf-25150fc41d5c)

**Comment by its-schmii** [comment]
Wow, I'm a little shocked by the lack of support.
But, I have to say I've been using the lib for a couple of weeks and just stumbled upon this.
I'm no whale, but I'll chip in
